### PR TITLE
Document CLI delete & chgrp

### PR DESCRIPTION
**Pull request by ximenesuk**
This PR adds page for the CLI `chgrp` and `delete` commands. The documentation should match and suitably illustrate the work in the PRs  https://github.com/openmicroscopy/openmicroscopy/pull/3725, https://github.com/openmicroscopy/openmicroscopy/pull/3824 and https://github.com/openmicroscopy/openmicroscopy/pull/3821

--no-rebase

### CODE REVIEW

**Comment by mtbc** [comment]
Staged at http://www.openmicroscopy.org/site/support/omero5.1-staging/users/cli/chgrp.html and http://www.openmicroscopy.org/site/support/omero5.1-staging/users/cli/delete.html.

File: omero/users/cli/chgrp.txt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    $ bin/omero chgrp 5 OriginalFile:101
+    $ bin/omero chgrp Group:5 Project:51
+    $ bin/omero chgrp ExperimenterGroup:5 Project:51
+    $ bin/omero chgrp lab_group Project:51
+
+In the first line, the original file with ID 101 will be moved to the group
+with ID 5. In the second, project 51 will be moved to group 5 including any
+datasets inside only that project and any images that are only contained within
+any moved datasets. If group 5 is named 'lab_group' then the last two lines
+will have the same effect as the second.
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: then the last line will have the same effect as the previous two ?

File: omero/users/cli/chgrp.txt [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+Moving lower level objects
+==========================
+
+To move objects below a specified top-level object the following form
+of the object specifier is used.::
+
+    $ bin/omero chgrp 5 Project/Dataset/Image:51
+
+Here the all of images under the project 51 would be moved. It is
+necessary to specify intermediate objects in the hierarchy and so::
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: *not* necessary

File: omero/users/cli/chgrp.txt [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+    $ bin/omero chgrp 5 Project/Image:51
+
+would have the same effect as the call above. Links can also be moved
+and so::
+
+$ bin/omero chgrp 5 Project/DatasetImageLink:51 Dataset/DatasetImageLink:53
+
+would effectively orphan all images under project 51 and dataset 53 that are
+not also under other datasets.
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: Does this really work? What happens? It's horrifying and I'm not sure we should be mentioning it!

File: omero/users/cli/chgrp.txt [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    This will move project 51 but not any datasets contained in that project.
+
+The two options can be used together.::
+
+     $ bin/omero chgrp 5 Project/Dataset:53 --exclude Image --include:FileAnnotation
+
+This will move any datasets under project 53, that are not otherwise
+contained elsewhere, excluding any images in those datasets but including
+and file annotations linked to the moved datasets. In this case the images
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: "any"

File: omero/users/cli/chgrp.txt [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+Further options
+^^^^^^^^^^^^^^^
+
+.. option:: --ordered
+
+    Move the objects in the order specified.
+
+    Normally all of the specified objects are grouped into a single move
+    command if possible. However, each object can be moved separately and
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: don't need "if possible" unless we know of any exceptions, "normally" is already weakening

File: omero/users/cli/chgrp.txt [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+::
+
+    $ bin/omero chgrp 5 OriginalFile:101
+    $ bin/omero chgrp Group:5 Project:51
+    $ bin/omero chgrp ExperimenterGroup:5 Project:51
+    $ bin/omero chgrp lab_group Project:51
+
+In the first line, the original file with ID 101 will be moved to the group
+with ID 5. In the second, project 51 will be moved to group 5 including any
+datasets inside only that project and any images that are only contained within
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: maybe, "... any images that are contained within moved datasets only"?

File: omero/users/cli/chgrp.txt [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    Linked objects that would ordinarily be moved can be excluded from the
+    move using the `--exclude` option.::
+
+        $ bin/omero chgrp 5 Project:51 --exclude Dataset
+
+    This will move project 51 but not any datasets contained in that project.
+
+The two options can be used together.::
+
+     $ bin/omero chgrp 5 Project/Dataset:53 --exclude Image --include:FileAnnotation
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: is the colon after `--include` intended?

File: omero/users/cli/chgrp.txt [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+The :command:`chgrp` command will remove entire graphs of objects based on the
+IDs of the topmost objects. The command can be modified to include the movement
+of objects that would, by default, be excluded or exclude objects that would,
+by default, be included using the :option:`--include` and :option:`--exclude`
+options.
+
+Unlike the :ref:`delete command <cli_delete>`, objects of the three annotation
+types, `FileAnnotation`, `TagAnnotation` and `TermAnnotation` are not moved
+by default when the objects to which they are linked are moved.
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: I don't think the that delete command would move them either. :smiley:

File: omero/users/cli/chgrp.txt (hunk 3 above)
Comment (ximenesuk) [inline omero/users/cli/chgrp.txt]: Yes, it is something you told me specifically about! I'm happy to not mention it though.

**Comment by mtbc** [comment]
Obviously a number of my comments on the "move" page apply correspondingly to the "delete" page.

File: omero/users/cli/chgrp.txt (hunk 8 above)
Comment (ximenesuk) [inline omero/users/cli/chgrp.txt]: Hmm, I guess I wanted to say that these three types are not treated specially as with delete. Maybe just not mentioning this at all is enough?

**Comment by ximenesuk** [comment]
@mtbc Thanks for the catches. I'll push some changes asap.

File: omero/users/cli/chgrp.txt (hunk 3 above)
Comment (ximenesuk) [inline omero/users/cli/chgrp.txt]: Ah, this is the `chgrp` page. I will remove from here - but is it still valid under `delete`?

Comment (mtbc) [inline omero/users/cli/chgrp.txt]: Yes, should be fine for `delete`.

File: omero/users/cli/chgrp.txt (hunk 8 above)
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: Maybe something like, "... are not *treated specially*, so they are not moved ..."

File: omero/users/cli/chgrp.txt [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+    Linked objects that would ordinarily be moved can be excluded from the
+    move using the `--exclude` option::
+
+        $ bin/omero chgrp 5 Project:51 --exclude Dataset
+
+    This will move project 51 but not any datasets contained in that project.
+
+The two options can be used together::
+
+     $ bin/omero chgrp 5 Project/Dataset:53 --exclude Image --include:FileAnnotation
```
Comment (mtbc) [inline omero/users/cli/chgrp.txt]: still doubtful about the colon in `--include:FileAnnotation`

Comment (ximenesuk) [inline omero/users/cli/chgrp.txt]: Sorry, that's a typo. I may have missed you commenting on it earlier. Now fixed.

**Comment by hflynn** [comment]
This looks good to me. I think we need to find a better way to signpost these docs but that's an issue I can think about next week!

**Comment by mtbc** [comment]
Fixes look good to me.

**Pull request merged**